app/models/mixins/authentication_mixin.rb

```diff
@@ -154,9 +164,11 @@ def authentication_status_ok?(type = nil)
   end
 
   def auth_user_pwd(type = nil)
-    cred = authentication_best_fit(type)
-    return nil if cred.nil? || cred.userid.blank?
-    [cred.userid, cred.password]
+    userid = authentication_userid(type)
+
+    return nil if userid.nil?
+
+    [userid, authentication_password(type).to_s]
```

Comment:
Note:  The `.to_s` is here to fix the specs, since there is a test that specifically looks for `''` when the password is `nil`.

I also DRY'd up some of the portion of the code that was duplicated here and in `authentication_component` (used by `authentication_userid`/`authentication_password`), so this is mostly trying to make this shared but still functionally equivalent.